dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -163,8 +170,28 @@ public static void main(String[] arguments)
         DependencyGraphBuilder.getStaticLinkageCheckDependencies(artifacts);
     List<DependencyPath> dependencyPaths = dependencyGraph.list();
 
+    // To remove duplicates on (groupId:artifactId) for dependency mediation
+    Map<String, String> keyToFirstArtifactVersion = Maps.newHashMap();
+
     for (DependencyPath dependencyPath : dependencyPaths) {
       Artifact artifact = dependencyPath.getLeaf();
+      String artifactVersion = artifact.getVersion();
+      // groupId:artifactId
+      String dependencyMediationKey = Artifacts.makeKey(artifact);
+      String firstArtifactVersionForKey = keyToFirstArtifactVersion.get(dependencyMediationKey);
+      if (firstArtifactVersionForKey != null
+          && !artifactVersion.equals(firstArtifactVersionForKey)) {
+        // Not adding this artifact if different version of the artifact (<groupId>:<artifactId> as
+        // key) is already in `multimap`.
+        // As `dependencyPaths` elements are in level order (breadth-first), this first-wins
+        // strategy follows Maven's dependency mediation.
+        // TODO(#309): add Gradle's dependency mediation
+        continue;
+      }
+      keyToFirstArtifactVersion.put(dependencyMediationKey, artifact.getVersion());
```

Comment:
Should all this happen after seeing whether `jarAbsolutePath` ends with ".jar", below?

Reply:
Updated.